draft-ietf-quic-multipath.md

```diff
@@ -790,32 +790,71 @@ In this example the client detects the network environment change
 or the quality of RTT or loss rate is becoming worse) and wants to close
 the initial path.
 
-In Figure {{fig-example-path-close}} the server's 1-RTT packets use DCID C1,
-which has a sequence number of 1, for the first path; the client's 1-RTT
-packets use DCID S2, which has a sequence number of 2.  For the second path,
-the server's 1-RTT packets use DCID C2, which has a sequence number of 2;
-the client's 1-RTT packets use CID S3, which has a sequence number of 3.
-Note that two paths use different packet number space.
-
-The client initiates the path closure for the path with ID 1 by sending
-a packet with an PATH_ABANDON frame. When the server received the
-PATH_ABANDON frame, it also sends an PATH_ABANDON frame in the next packet.
-Afterwards the connection IDs in both directions can be retired
-using the RETIRE_CONNECTION_ID frame.
+{{fig-example-path-close1}} illustrates an example of path closing when both
+the client and the server use non-zero-length CIDs. For the first path, the
+server's 1-RTT packets use DCID C1, which has a sequence number of 1; the
+client's 1-RTT packets use DCID S2, which has a sequence number of 2. For the
+second path, the server's 1-RTT packets use DCID C2, which has a sequence
+number of 2; the client's 1-RTT packets use CID S3, which has a sequence number
+of 3. Note that two paths use different packet number spaces. In this case, the
+client is going to close the first path. It identifies the path by the sequence
+number of the received packet's DCID over that path (path identifier type
+0x00), hence using a path_id 1. In reply, the server confirms the path using
+the sequence number of the received packet's DCID over that path (path
+identifier type 0x00), which corresponds to a path_id 2. Both the client and
+the server can close the path after receiving the RETIRE_CONNECTION_ID frame
+for that path.
 
 ~~~
 Client                                                      Server
 
 (client tells server to abandon a path)
-1-RTT[X]: DCID=S2 PATH_ABANDON[path_id=1]->
+1-RTT[X]: DCID=S2 PATH_ABANDON[path_id_type=0, path_id=1]->
```

Comment:
I'm still puzzled by this example. If I read the definition of the path_id_type in the frame definition, it says

```
Type 0: Refer to the connection identifier used by the sender of the control frame when sending
data over the specified path. This method SHOULD be used if this connection identifier is
non-zero length. This method MUST NOT be used if this connection identifier is zero-length.
```
As the sender of the frame is the client, it should thus refer to the connection identifier used by the client to send packets on the first path (DCID S2), thus the path_id should be 2.

Reply:
Yes, I think the text says it correctly, only the figure is wrong. Good catch!

Reply:
In this case, the path_id is the sequence number of the CID issued by the client for this path and thus, it is 1.